zipline/src/jvmTest/kotlin/app/cash/zipline/ConsoleTest.kt

```diff
@@ -20,29 +20,28 @@ import assertk.assertThat
 import assertk.assertions.contains
 import assertk.assertions.isEqualTo
 import assertk.assertions.matches
-import java.util.concurrent.LinkedBlockingDeque
 import java.util.logging.Handler
 import java.util.logging.Level
 import java.util.logging.LogRecord
 import java.util.logging.Logger
 import kotlin.test.assertEquals
 import kotlin.test.assertNull
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runTest
 import org.junit.After
 import org.junit.Before
-import org.junit.Rule
 import org.junit.Test
 
 class ConsoleTest {
-  @Rule @JvmField
-  val ziplineTestRule = ZiplineTestRule()
-  private val dispatcher = ziplineTestRule.dispatcher
+  private val dispatcher = StandardTestDispatcher()
   private val zipline = Zipline.create(dispatcher)
 
-  private val logRecords = LinkedBlockingDeque<LogRecord>()
+  private val logRecords = Channel<LogRecord>(UNLIMITED)
```

Comment:
This is half the problem. Calling `take()` blocks the dispatcher and prevents Zipline from posting a coroutine to append to the queue. A channel allows the test to suspend while waiting.